File: src/main/java/ru/citeck/idea/references/resolvers/PsiLiteralExpressionValueResolver.java
package ru.citeck.idea.references.resolvers;

import com.intellij.psi.PsiLiteralExpression;
import ru.citeck.idea.references.PsiElementValueResolver;

public class PsiLiteralExpressionValueResolver implements PsiElementValueResolver<PsiLiteralExpression> {

    @Override
    public Class<PsiLiteralExpression> getPsiElementType() {
        return PsiLiteralExpression.class;
    }

    @Override
    public Object getValue(PsiLiteralExpression psiElement) {
        return psiElement.getValue();
    }
}

File: src/main/kotlin/ru/citeck/idea/references/resolvers/PsiLiteralExpressionValueResolver.kt
package ru.citeck.idea.references.resolvers

import com.intellij.psi.PsiLiteralExpression
import ru.citeck.idea.references.PsiElementValueResolver

class PsiLiteralExpressionValueResolver : PsiElementValueResolver<PsiLiteralExpression> {

    override fun getPsiElementType(): Class<PsiLiteralExpression> {
        return PsiLiteralExpression::class.java
    }

    override fun getValue(psiElement: PsiLiteralExpression): Any? {
        return psiElement.value
    }
}
